client/me/billing-history/upcoming-charges-table.jsx

```diff
@@ -1,62 +1,63 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal Dependencies
  */
-var	purchasesPaths = require( 'me/purchases/paths' ),
-	TransactionsTable = require( './transactions-table' );
+import purchasesPaths from 'me/purchases/paths';
+import TransactionsTable from './transactions-table';
+import { getSitesForUpcomingTransactions } from 'state/selectors';
+
+class UpcomingChargesTable extends Component {
+	static propTypes = {
+		// Computed props
+		sites: PropTypes.object.isRequired,
+	}
 
-module.exports = React.createClass( {
+	renderTransaction = ( transaction ) => {
+		const { translate } = this.props;
+		const site = this.props.sites[ Number( transaction.blog_id ) ];
 
-	displayName: 'UpcomingChargesTable',
+		if ( ! site ) {
+			return null;
```

Comment:
Also @gwwar in a9c26b7139c5c4e7f44a277751c2dc7406349121 I've altered the selector to work with slugs only, as the entire site objects are not necessary.